taemin/plugins/voxpopuli/plugin.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+# -*- coding: utf8 -*-
+
+"""
+    Kick Vote Plugin
+"""
+
+import time
+from threading import Thread
+
+from taemin import plugin
+
+class TaeminVoxPopuli(plugin.TaeminPlugin):
+    """ Vox Populi Plugin """
+
+    helper = {
+        "voxpopuli": "Vote to kick someone: '!voxpopuli [nick] [vote duration (s)]",
+    }
+
+    def __init__(self, taemin):
+        plugin.TaeminPlugin.__init__(self, taemin)
+        self.states = {}
+
+    def on_pubmsg(self, msg):
+        if msg.key not in self.helper:
+            return
+
+        chan = msg.chan.name
+
+        if chan not in self.states:
+            self.states[chan] = VoxWait(self, chan)
+
+        if msg.key == "voxpopuli":
+            self.swap_state(chan, self.states[chan].on_msg(msg.value, msg.user.name))
+
+    def on_timeout(self, chan):
+        """ Called once the vote is finished """
+
+        if chan not in self.states:
+            return
+
+        self.swap_state(chan, self.states[chan].on_timeout())
+
+    def swap_state(self, chan, new_state):
```

Comment:
I don't really get the point of this function, I think it should be removed, it looks like a remain of a previous try to have it change between VoxWait and VoxVote but right now it should just be a dictionary assignment.